lib/rouge/lexers/ecl.rb

```diff
@@ -0,0 +1,140 @@
+      )
+
+      keywords = %w(
+        and or in not all any as from 
+        atmost before beginc best between case const counter 
+        csv descend embed encrypt end endc endembed endmacro 
+        enum except exclusive expire export extend fail few 
+        first flat full function functionmacro group heading hole 
+        ifblock import joined keep keyed last left limit 
+        load local locale lookup macro many maxcount maxlength 
+        _token module interface named nocase noroot noscan nosort 
+        of only opt outer overwrite packed partition penalty 
+        physicallength pipe quote record repeat return right rows 
+        scan self separator service shared skew skip sql 
+        store terminator thor threshold token transform trim type 
+        unicodeorder unsorted validate virtual whole wild within xml 
+        xpath after cluster compressed compression default encoding escape 
+        fileposition forward grouped inner internal linkcounted literal lzw 
+        mofn multiple namespace wnotrim noxpath onfail prefetch retry 
+        rowset scope smart soapaction stable timelimit timeout unordered 
+        unstable update use width
+      )
+
+      functions = %w(
+        abs acos aggregate allnodes apply ascii asin asstring 
+        atan _token ave case catch choose choosen choosesets 
+        clustersize combine correlation cos cosh count covariance cron 
+        dataset dedup define denormalize dictionary distribute distributed distribution 
+        ebcdic enth error evaluate event eventextra eventname exists 
+        exp failcode failmessage fetch fromunicode fromxml getenv getisvalid 
+        global graph group hash hashcrc having httpcall httpheader 
+        if iff index intformat isvalid iterate join keyunicode 
+        length library limit ln local log loop map 
+        matched matchlength matchposition matchtext matchunicode max merge mergejoin 
+        min nofold nolocal nonempty normalize parse pipe power 
+        preload process project pull random range rank ranked 
+        realformat recordof regexfind regexreplace regroup rejected rollup round 
+        roundup row rowdiff sample set sin sinh sizeof 
+        soapcall sort sorted sqrt stepped stored sum table 
+        tan tanh thisnode topn tounicode toxml transfer transform 
+        trim truncate typeof ungroup unicodeorder variance which workunit 
+        xmldecode xmlencode xmltext xmlunicode apply assert build buildindex 
+        evaluate fail keydiff keypatch loadxml nothor notify output 
+        parallel sequential soapcall wait
+      )
+
+      class1 = %w(
+        file date str math metaphone metaphone3 uni audit blas system
+      )
+
+      class2 = %w(
+        debug email job log thorlib util workunit
+      )
+
+      state :string_character_escape do
+        rule %r/\\\\(x\\h{2}|[0-2][0-7]{,2}|3[0-6][0-7]?|37[0-7]?|[4-7][0-7]?|.|$)/, Text
```

Comment:
Is the double backslash correct? So an escaped string character could be `\\x\h\h`? Separately, a better token for this rule would be `String::Escape`.

Reply:
You are right. I was using another lexer as an example and it's regex patterns are in double quotes.